WordPress/gutenberg
# Categories List Block: Match the markup on the block editor to the front end

Fix: #42984

## What?
This PR makes the markup on the block editor the same as on the front editor in order to keep the markup consistent.

## Why?
If class names and makups are different between the editor and the front end, it will be difficult to use a common selector for styling.

Also, I could not find any reason for different markup on the editor.

## How?
In order to get closer to the frontend markup, I have taken the following actions:

-  Change the block tag according to its display status (i.e., when displayed in a list, it becomes `ul` tag)
- Remove unused elements and class names on the front end (`.wp-block-categories__list`, `.wp-block-categories__list-level-{level}`, `.wp-block-categories__post-count`, `.wp-block-categories__dropdown`）
- Add empty option tag ("Select Category") to dropdown
- Add missing class names (`li.cat-item,cat-item-{id}`, `option.level-{level}`)



## Testing Instructions
- Insert the Categories List Block.
- Toggle the option variations and confirm that the frontend and block editor markup is nearly identical for both variations.

## Screenshots or screencast

Below is a comparison of the block editor and front-end markup for one variation.

### Frontend

![frontend](https://user-images.githubusercontent.com/54422211/187031083-6352c0d7-c330-45af-96b0-1049eee5e45b.png)

### Block Editor

![block-editor](https://user-images.githubusercontent.com/54422211/187031073-cb9c2524-2dd4-4706-981f-549e3db8335f.png)
